sqlserver/datadog_checks/sqlserver/activity.py

```diff
@@ -0,0 +1,226 @@
+        cursor.execute(CONNECTIONS_QUERY)
+        columns = [i[0] for i in cursor.description]
+        # construct row dicts manually as there's no DictCursor for pyodbc
+        rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        self.log.debug("loaded sql server current connections len(rows)=%s", len(rows))
+        return rows
+
+    def _get_activity(self, cursor):
+        self.log.debug("collecting sql server activity")
+        extra_query_args = self._get_extra_activity_query_args()
+        query = ACTIVITY_QUERY.format(extra_query_args=extra_query_args)
+        self.log.debug("Running query [%s]", query)
+        cursor.execute(query)
+        columns = [i[0] for i in cursor.description]
+        # construct row dicts manually as there's no DictCursor for pyodbc
+        rows = [dict(zip(columns, row)) for row in cursor.fetchall()]
+        return rows
+
+    def _get_extra_activity_query_args(self):
+        extra_query_args = (
+            " AND NOT (sess.status = 'sleeping' AND DATEDIFF(second, at.transaction_begin_time, GETDATE()) < {})"
+        ).format(self.collection_interval)
+        # order results by tx begin time to get longest running transactions first.
+        extra_query_args = extra_query_args + " ORDER BY at.transaction_begin_time ASC"
+        return extra_query_args
+
+    def _normalize_queries_and_filter_rows(self, rows, max_bytes_limit):
+        normalized_rows = []
+        estimated_size = 0
+        for row in rows:
+            try:
+                obfuscated_statement = datadog_agent.obfuscate_sql(row['text'])
+                row['query_signature'] = compute_sql_signature(obfuscated_statement)
+            except Exception as e:
+                # obfuscation errors are relatively common so only log them during debugging
+                self.log.debug("Failed to obfuscate query: %s", e)
+                obfuscated_statement = "ERROR: failed to obfuscate"
+            row = self._sanitize_row(row, obfuscated_statement)
+            estimated_size += self._get_estimated_row_size_bytes(row)
+            if estimated_size > max_bytes_limit:
+                # query results are ORDER BY transaction_begin_time ASC
+                # so once we hit the max bytes limit, return
+                return normalized_rows
+            normalized_rows.append(row)
+        return normalized_rows
+
+    @staticmethod
+    def _sanitize_row(row, obfuscated_statement):
+        row = {key: val for key, val in row.items() if key not in dm_exec_requests_exclude_keys and val is not None}
+        row['text'] = obfuscated_statement
+        if 'query_hash' in row:
+            row['query_hash'] = _hash_to_hex(row['query_hash'])
+        if 'query_plan_hash' in row:
+            row['query_plan_hash'] = _hash_to_hex(row['query_plan_hash'])
+        return row
+
+    @staticmethod
+    def _get_estimated_row_size_bytes(row):
+        return len(str(row))
```

Comment:
Why convert the object to a string representation vs dump to json? They're not quite the same thing.

Reply:
It ends up being about the same size in the end, but we would be doing a json serialization call on every row for nothing since we want to wait until the end to do full serialization